impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java

```diff
@@ -38,55 +37,81 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-/** Tests for {@link TagContextImpl} and {@link TagContextBuilderImpl}. */
-// TODO(sebright): Add more tests once the API is finalized.
+/**
+ * Tests for {@link TagContextImpl} and {@link TagContextBuilderImpl}.
+ *
+ * <p>Tests for {@link TagContextBuilderImpl#buildScoped()} are in {@link ScopedTagContextsTest}.
+ */
 @RunWith(JUnit4.class)
 public class TagContextImplTest {
   private final Tagger tagger = new TaggerImpl(new CurrentTaggingState());
 
-  private static final TagKey KS1 = TagKey.create("k1");
-  private static final TagKey KS2 = TagKey.create("k2");
+  private static final TagKey K1 = TagKey.create("k1");
+  private static final TagKey K2 = TagKey.create("k2");
 
   private static final TagValue V1 = TagValue.create("v1");
   private static final TagValue V2 = TagValue.create("v2");
 
   @Rule public final ExpectedException thrown = ExpectedException.none();
 
   @Test
-  public void testSet() {
-    TagContext tags = tagger.emptyBuilder().put(KS1, V1).build();
-    assertThat(tagContextToList(tagger.toBuilder(tags).put(KS1, V2).build()))
-        .containsExactly(Tag.create(KS1, V2));
-    assertThat(tagContextToList(tagger.toBuilder(tags).put(KS2, V2).build()))
-        .containsExactly(Tag.create(KS1, V1), Tag.create(KS2, V2));
+  public void getTags_empty() {
+    TagContextImpl tags = new TagContextImpl(ImmutableMap.<TagKey, TagValue>of());
+    assertThat(tags.getTags()).isEmpty();
+  }
+
+  @Test
+  public void getTags_nonEmpty() {
+    TagContextImpl tags = new TagContextImpl(ImmutableMap.of(K1, V1, K2, V2));
+    assertThat(tags.getTags()).containsExactly(K1, V1, K2, V2);
+  }
+
+  @Test
+  public void put_newKey() {
```

Comment:
Add tests on putting null key and null value?

Reply:
Good catch!